### PR TITLE
Enable config to expose Envoy's X-RateLimit headers

### PR DESCRIPTION
**Pull request by jbohanon**
# Description

This new feature exposes API options to enable X-RateLimit headers in Envoy/rate-limiter config. The related rate-limiter PR for implementing the return of the headers is [here](https://github.com/solo-io/rate-limiter/pull/133)

# Context

Users would like some response headers to indicate which rate limit policy has been exceeded and how long until that policy is reset.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/api/v1/enterprise/options/ratelimit/ratelimit.proto [hunk 1]
```diff
@@ -25,6 +25,8 @@ message Settings {
 
   bool deny_on_fail = 3;
 
+  bool enable_x_ratelimit_headers = 4;
```
Comment (kdorosh) [inline projects/gloo/api/v1/enterprise/options/ratelimit/ratelimit.proto:28]: nit; can we add some comments to the user-facing api here for our generated docs?

Comment (jbohanon) [reply projects/gloo/api/v1/enterprise/options/ratelimit/ratelimit.proto:28]: Great point; I was just looking into updating the guide as well, so I'll put a comment in with that commit

File: projects/gloo/pkg/plugins/ratelimit/util.go [hunk 2]
```diff
@@ -15,6 +15,7 @@ func GenerateEnvoyConfigForFilterWith(
 	stage uint32,
 	timeout *duration.Duration,
 	denyOnFail bool,
+	enableXRatelimitHeaders bool,
```
[the remark this answers is not in the document]
Comment (jbohanon) [reply projects/gloo/pkg/plugins/ratelimit/util.go:18]: Yes! thanks for catching; I'll update changelog

**Comment by jbohanon** [comment]
/kick build bot didn't restart on changelog-only push

**Review by jackstine — APPROVED** [review]
🔥

**Pull request merged**